lib/matplotlib/tests/test_colorbar.py

```diff
@@ -622,3 +622,51 @@ def test_colorbar_int(clim):
     im = ax.imshow([[*map(np.int16, clim)]])
     fig.colorbar(im)
     assert (im.norm.vmin, im.norm.vmax) == clim
+
+
+def test_anchored_cbar_position_using_specgrid():
+    data = np.arange(1200).reshape(30, 40)
+    levels = [0, 200, 400, 600, 800, 1000, 1200]
+    shrink = 0.5
+    anchor_y = 0.3
+    # vertival
+    fig, ax = plt.subplots()
+    cs = ax.contourf(data, levels=levels)
+    cbar = plt.colorbar(
+            cs, ax=ax, use_gridspec=True,
+            orientation='vertical', anchor=(1, anchor_y), shrink=shrink)
+
+    # y1: the top of ax, y0: the bottom of ax, p0: the y postion of anchor
+    # cy1 : the top of colorbar ax, cy0: the bottom of colorbar ax
+    y1 = ax.get_position().y1
+    y0 = ax.get_position().y0
+    p0 = (y1 - y0) * anchor_y + y0
+    cy1 = cbar.ax.get_position().y1
+    cy0 = cbar.ax.get_position().y0
+
+    assert np.isclose(
+            [cy1, cy0],
+            [y1 * shrink + (1 - shrink) * p0, p0 * (1 - shrink) + y0 * shrink]
+            ).all()
```

Comment:
Can use `assert_allclose` from `np.testing` instead.

Reply:
Thank you for the suggestion, now using `np.testing.assert_allclose`